plc4j/utils/opm/src/main/java/org/apache/plc4x/java/opm/PlcEntityManager.java

```diff
@@ -0,0 +1,380 @@
+                    setField(o.getClass().getSuperclass(), o, response, fieldName);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    throw new PlcRuntimeException(e);
+                }
+            }
+        } catch (Exception e) {
+            throw new OPMException("Problem during processing", e);
+        }
+    }
+
+    // TODO: why isn't o used?
+    private Object fetchValueForGetter(Object o, Method m) throws OPMException {
+        String s = m.getName().substring(3);
+        // First char to lower
+        String variable = s.substring(0, 1).toLowerCase().concat(s.substring(1));
+        LOGGER.trace("Looking for field with name {} after invokation of getter {}", variable, m.getName());
+        PlcField annotation;
+        try {
+            annotation = m.getDeclaringClass().getDeclaredField(variable).getDeclaredAnnotation(PlcField.class);
+        } catch (NoSuchFieldException e) {
+            throw new OPMException("Unable to identify field annotated field for call to " + m.getName(), e);
+        }
+        PlcEntity plcEntity = m.getDeclaringClass().getAnnotation(PlcEntity.class);
+        try (PlcConnection connection = driverManager.getConnection(plcEntity.value())) {
+            // Catch the exception, if no reader present (see below)
+            PlcReader plcReader = connection.getReader().orElseThrow(IllegalStateException::new);
+
+            // Assume to do the query here...
+            PlcReadRequest request = plcReader.readRequestBuilder()
+                .addItem(m.getName(), annotation.value())
+                .build();
+
+            PlcReadResponse<?> response;
+            response = getPlcReadResponse(plcReader, request);
+
+            return getTyped(m.getReturnType(), response, m.getName());
+        } catch (ClassCastException e) {
+            throw new OPMException("Unable to return response as suitable type", e);
+        } catch (Exception e) {
+            throw new OPMException("Problem during processing", e);
+        }
+    }
+
+    /**
+     * Tries to set a response Item to a field in the given object.
+     * This is one by looking for a field in the class and a response item
+     * which is equal to the given fieldName parameter.
+     *
+     * @param o         Object to set the value on
+     * @param response  Response to fetch the response from
+     * @param fieldName Name of the field in the object and the response
+     * @throws NoSuchFieldException
+     * @throws IllegalAccessException
+     */
+    private void setField(Class<?> clazz, Object o, PlcReadResponse<?> response, String fieldName) throws NoSuchFieldException, IllegalAccessException {
+        Field field = clazz.getDeclaredField(fieldName);
+        field.setAccessible(true);
+        try {
+            field.set(o, getTyped(field.getType(), response, fieldName));
+        } catch (ClassCastException e) {
+            // TODO should we simply fail here?
```

Comment:
think so as this can be checked beforehand and so an simple fail is sufficient here.

Reply:
In theory this should be an IllegalStateException, as this should never happen, or?